File: problems/#230/v2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        vector<int> v;
        return BST2Vector(root, v)[k - 1];
    }

    vector<int> BST2Vector(TreeNode* n, vector<int> &v) {
        if (n == NULL) {
            return v;
        }
        BST2Vector(n->left, v);
        v.push_back(n->val);
        BST2Vector(n->right, v);
        return v;
    }
};

File: problems/#2415/v2.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func reverseOddLevels(root *TreeNode) *TreeNode {
    cur, nxt, lvl := make([]*TreeNode, 0), make([]*TreeNode, 0), 0
    cur = append(cur, root)
    for cur[0].Left != nil {
        for _, node := range cur {
            nxt = append(nxt, node.Left)
            nxt = append(nxt, node.Right)
        }
        lvl++
        if lvl % 2 == 1 {
            for l, r := 0, len(nxt) - 1; l < r; l, r = l + 1, r - 1 {
                nxt[l].Val, nxt[r].Val = nxt[r].Val , nxt[l].Val 
            }
        }
        cur, nxt = nxt, make([]*TreeNode, 0)
    }
    return root
}